### ISSUE TITLE
Themes and plugins gallery - Web

### ISSUE THREAD

**Issue by JosefNemec** [JosefNemec/Playnite#593]
Establish web based gallery for themes, plugins and scripts. I would preferably use some already made framework if available.

Must have features:
- File versioning. Uploaded items must be stored under specific version and user must be able to download older versions.
- Metadata support. System must be able to assign custom metadata to uploaded items, this is needed for extensions to report supported Playnite SDK version for #298
- Rating. Users can rate uploaded items.
- Integration with forums. The same accounts should be shared with #430 between gallery and forums.

Nice to have:
- API. If possible chosen galley should provide some sort of API so we don't have to write our own for #298 instead of accessing DB directly. 

Basically something like NuGet gallery with ability to rate items and better support for item descriptions (add screenshots etc.)

Integration with Playnite application itself will be done afterwards in #298

**Comment by JosefNemec** [comment]
After doing some research I've found that there's no ready made solution that we could use (or at least was future proof for upcoming Playnite features). We will have to implement our own.

**Comment by JosefNemec** [comment]
The temporary solution for now will be opening special sub-forums on official forum #430 until the gallery is complete.

**Comment by salty-biscuits** [comment]
I can make the frontend for it, as I am unsure what backend language you're using right now and I can't be arsed to learn another language.

**Comment by JosefNemec** [comment]
Thanks for the offer, but I've postponed this indefinitely, because there are more important and more useful features to be added.  I'll be launching new forums soon that should act as extension gallery in the mean time.

**Comment by benblo** [comment]
As a temporary 0-cost alternative, might I suggest a simple public-wiki-style .md file, [like this](https://github.com/nothings/single_file_libs)?

**Comment by scowalt** [comment]
> As a temporary 0-cost alternative, might I suggest a simple public-wiki-style .md file, [like this](https://github.com/nothings/single_file_libs)?

I started [a repository to do exactly that](https://github.com/scowalt/PlayniteExtensionList), since it looks like there hasn't been any progress on this issue. Feel free to contribute any extensions I might have missed! Extensions and themes are a bit difficult to find a curate, which underlines the need for some resolution of this issue.

**Comment by JosefNemec** [comment]
There will be add-on database integrated into Playnite 9, similar to how how winget does it (add-on manifests stored in special repository with custom add-on broswer in Playnite itself).

**Comment by scowalt** [comment]
Awesome, that sounds like a great solution. Will the special extension repository be publically accessible in .git land, or will the implementation be something else?

**Comment by JosefNemec** [comment]
It will be literally a git repository. As I said, very similar to how [winget](https://github.com/microsoft/winget-pkgs/tree/master/manifests) does it.

**Comment by scowalt** [comment]
I assume this issue is resolved now with the release of Playnite 9?

**Comment by JosefNemec** [comment]
No it wasn't, we still don't have a web page to browse add-ons.
